StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1206DeclarationKeywordsMustFollowOrder.cs

```diff
@@ -54,7 +79,177 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+                    break;
+                case SyntaxKind.StaticKeyword:
+                    result = ModifierType.Static;
+                    break;
+                case SyntaxKind.VirtualKeyword:
+                case SyntaxKind.AbstractKeyword:
+                case SyntaxKind.OverrideKeyword:
+                case SyntaxKind.ExternKeyword:
+                case SyntaxKind.UnsafeKeyword:
+                case SyntaxKind.NewKeyword:
+                case SyntaxKind.SealedKeyword:
+                case SyntaxKind.ReadOnlyKeyword:
+                case SyntaxKind.VolatileKeyword:
+                case SyntaxKind.FixedKeyword:
+                case SyntaxKind.ConstKeyword:
+                case SyntaxKind.AsyncKeyword:
+                case SyntaxKind.PartialKeyword:
+                    result = ModifierType.Other;
+                    break;
+            }
+
+            return result;
+        }
+
+        private static int CompareModifiersType(ModifierType first, ModifierType second)
+        {
+            const int lessThan = -1;
+            const int greaterThan = 1;
+
+            var result = 0;
+
+            if (first == second)
+            {
+                result = 0;
+            }
+            else if (first == ModifierType.None)
+            {
+                result = lessThan;
+            }
+            else if (second == ModifierType.None)
+            {
+                result = greaterThan;
+            }
+            else if (first == ModifierType.Access && (second == ModifierType.Static || second == ModifierType.Other))
+            {
+                result = lessThan;
+            }
+            else if (first == ModifierType.Static && second == ModifierType.Other)
+            {
+                result = lessThan;
+            }
+            else if (first == ModifierType.Static && second == ModifierType.Access)
+            {
+                result = greaterThan;
+            }
+            else if (first == ModifierType.Other && (second == ModifierType.Static || second == ModifierType.Access))
+            {
+                result = greaterThan;
+            }
+
+            return result;
+        }
+
+        private static string GetModifierTypeText(ModifierType modifierType)
+        {
+            var result = string.Empty;
+
+            switch (modifierType)
+            {
+                case ModifierType.Access:
+                    result = "access modifier";
+                    break;
+                case ModifierType.Static:
+                    result = "static";
+                    break;
+                case ModifierType.Other:
+                    result = "other";
+                    break;
+            }
+
+            return result;
         }
+
+        private static bool AccessOrStaticModifierNotFollowingOtherModifier(ModifierType current, ModifierType previous) => (current == ModifierType.Access || current == ModifierType.Static) && previous != ModifierType.Other;
```

Comment:
If the expression is more than one token, this syntax will probably make readability *worse*. I can't tell where the signature ends and the body begins when I'm glancing at this.